apel/common/datetime_utils.py

```diff
@@ -47,10 +47,16 @@ def parse_timestamp(datetime_string):
     '''
     Parse timestamp encoded as a string in various forms.  Return
     a TZ-unaware datetime object, which is in UTC.
-    
+
     If timezone information is not present in the string, it
-    assumes that the timezone is UTC.  
+    assumes that the timezone is UTC.
     '''
+
+    # If format is YYYYMMDDhhmmssZ (no T) add time designator so it is strictly
+    # compliant with ISO 8601. (Some versions of iso8601 module are fussy.)
+    if re.match(r'[0-9]{14}Z', datetime_string) is not None:
```

Comment:
Shall we use regexp like:
``r^[0-9]{14}Z`` instead of the one above? It currently accepts every text containing YYYYMMDDhhmmssZ but it should accept only one starting with date.